### PR TITLE
Quickstart 10 - JSF/JPA - Tasks JSF

### PR DESCRIPTION
**Pull request by lfryc**

### CODE REVIEW

**Comment by pmuir** [comment]
Lukas, could you rebase this?

```
git fetch upstream && git rebase upstream/master
```

and then push it (with the -f option).

**Comment by lfryc** [comment]
Hey Pete, I have sqaushed the commits, since the branch were really chaos.
I think it's appropriate in this case.

**Comment by pmuir** [comment]
Awesome, that would have been my next request ;-)

**Comment by sgilda** [comment]
Hi Lukas,
I tested the quickstart and it works fine. I'm still perplexed by the Arquillian tests though :-). I can see that all the tests passed, but don't know what it's doing under the covers. Are the XML files in the surefire-reports directory the output of the tests? Do we assume most developers know what to look for when running Arquillian tests?
I found a few minor differences in version numbers in your pom.xml file:
-- Your pom is the only one with a jbossas.version (<jbossas.version>7.1.0.Beta1b</jbossas.version>). Is this something new?
-- Your pom has version 2.10 for the maven-surefire-plugin with a comment that versions prior to 2.9 contain a bug. The checklist shows 2.4.3 for the current version. That's a big discrepancy. Is that correct?
--- You define the jbossas.version value as 7.1.0.Beta1b and that value is used for the jboss-as-maven-plugin, however ,the checklist shows it should be 7.1.0.CR1.
Other than that, it looks good!

Pete, did you do a code review? 
Thanks,
Sande

**Comment by pmuir** [comment]
On 13 Feb 2012, at 15:45, sgilda wrote:

> Hi Lukas,
> I tested the quickstart and it works fine. I'm still perplexed by the Arquillian tests though :-). I can see that all the tests passed, but don't know what it's doing under the covers. Are the XML files in the surefire-reports directory the output of the tests? Do we assume most developers know what to look for when running Arquillian tests?
> I found a few minor differences in version numbers in your pom.xml file:
> -- Your pom is the only one with a jbossas.version (<jbossas.version>7.1.0.Beta1b</jbossas.version>). Is this something new?

This should be replaced by usage of the javaee-6.0-with-tools BOM that Karel did.

**Comment by lfryc** [comment]
Hey,

I will refactor sample to use javaee-6.0-with-tools.

Btw, are you fine with including the Arquillian tests from QS-8 in this sample?
I'm considering it useful having tests in place side by side with view layer.

Other than that, I will possibly merge all the changes from sample QS-8 [1],
since my sample is based on that one,
and we would add more info about running and reviewing Arquillian tests.

[1] https://github.com/jbossas/quickstart/pull/91

**Comment by pmuir** [comment]
All sounds good.

**Comment by lfryc** [comment]
Okay, all comments above should be reflected.

Could you please take another look?

**Comment by sgilda** [comment]
Hi Lukas,
I see that you added a jsf to this example. That's a really nice addition!

I don't see anything in the README to indicate that there is a GUI to this quickstart. Could you add a section before "Running the Arquillian tests", something like the following:

## Deploying the application

First you need to start JBoss AS 7 (or JBoss Enterprise Application Platform 6). To do this, run:

```
    $JBOSS_HOME/bin/standalone.sh
```

or if you are using Windows:

```
    $JBOSS_HOME/bin/standalone.bat
```

To deploy the application, you first need to produce the archive using the following Maven goal:

```
    mvn package
```

You can now deploy the artifact to JBoss AS by executing the following command:

```
    mvn jboss-as:deploy
```

This will deploy `target/jboss-as-tasks-jsf.war` to the running instance of the AS.

The application will be running at the following URL http://localhost:8080/jboss-as-tasks-jsf.

To undeploy from JBoss AS, run this command:

```
    mvn jboss-as:undeploy
```

**Comment by lfryc** [comment]
Hello Sande, I have updated the instructions.

**Comment by sgilda** [comment]
One thing I missed (and I think we're missing from most of the quickstart instructions) is to navigate to the quickstart directory. I'm assume they should know that, but it doesn't hurt to be specific.

Did you want to add that? Or, I can go through them all and add that later if you prefer.

Pete, I'm good with this one if you are.

File: tasks-jsf/pom.xml [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+                            <skip>true</skip>
+                        </configuration>
+                    </plugin>
+                    <!-- The JBoss AS plugin deploys your war to a local JBoss
+                        AS container -->
+                    <!-- To use, run: mvn package jboss-as:deploy -->
+                    <plugin>
+                        <groupId>org.jboss.as.plugins</groupId>
+                        <artifactId>jboss-as-maven-plugin</artifactId>
+                        <version>7.1.0.Final</version>
```
Comment (pmuir) [inline tasks-jsf/pom.xml]: Why is this in the profile? Normally it's in the main plugins section.

File: tasks-jsf/README.md [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+
+
+What is it?
+-----------
+
+This is your project! It's a sample Maven 3 project to help you
+get your foot in the door developing with Java EE 6 on JBoss AS 7 or JBoss EAP 6. 
+This project is setup to allow you to use JPA 2.0 persistence with JSF 2.0 as view layer.
+
+The theme of this application is simple Task management with authentication.
```
Comment (pmuir) [inline tasks-jsf/README.md]: I think the word "authentication" is wrong - we don't authenticate the users at all, we trust them to be who they say they are!

How about we just call it "simple log in".

File: tasks-jsf/src/main/java/org/jboss/as/quickstarts/tasksJsf/web/AuthController.java [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+
+/**
+ * Provides authentication operations with current user store: {@link Authentication}.
+ *
+ * @author Lukas Fryc
+ *
+ */
+@Named("auth")
+@RequestScoped
+public class AuthController {
```
Comment (pmuir) [inline tasks-jsf/src/main/java/org/jboss/as/quickstarts/tasksJsf/web/AuthController.java]: Avoiding contractions is a good idea - we've avoided it so far :-)

File: tasks-jsf/src/main/java/org/jboss/as/quickstarts/tasksJsf/web/NavigationController.java [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+import javax.inject.Inject;
+import javax.inject.Named;
+
+/**
+ * Redirects user to another view when he requests page without sufficient permissions.
+ *
+ * @author Lukas Fryc
+ *
+ */
+@Named("nav")
```
Comment (pmuir) [inline tasks-jsf/src/main/java/org/jboss/as/quickstarts/tasksJsf/web/NavigationController.java]: Another contraction

File: tasks-jsf/src/main/java/org/jboss/as/quickstarts/tasksJsf/web/TaskListBean.java [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+ *
+ * <p>
+ * Delegates to {@link TaskDao} for persistence operations.
+ * </p>
+ *
+ * @author Lukas Fryc
+ */
+@Named("taskList")
+@RequestScoped
+public class TaskListBean implements TaskList, Serializable {
```
Comment (pmuir) [inline tasks-jsf/src/main/java/org/jboss/as/quickstarts/tasksJsf/web/TaskListBean.java]: Perhaps explain the CDI pattern you are using here a bit.

**Comment by pmuir** [comment]
A few minor comments, otherwise good.

**Comment by lfryc** [comment]
Hi Pete, I have incorporated changes you suggested and rebased on upstream.

**Comment by sgilda** [comment]
I've tested this and it works great.
Pete, do you need to review the code changes?

**Comment by pmuir** [comment]
Lukas, can you rebase onto my work on the tasks quickstart - pull 153. I'll then do a last review and we can merge.

**Comment by lfryc** [comment]
Hey Pete, I have rebased manually your work on pull 153 (commit "Various") and fixed various issue to make it work fluently.

Could you give it a spin?

File: tasks-jsf/src/main/java/org/jboss/as/quickstarts/tasksJsf/AuthController.java [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+    /**
+     * Logs current user out and ends the current conversation.
+     */
+    public void logout() {
+        authentication.setCurrentUser(null);
+        conversation.end();
+    }
+
+    /**
+     * Returns true if user is logged
```
Comment (pmuir) [inline tasks-jsf/src/main/java/org/jboss/as/quickstarts/tasksJsf/AuthController.java]: Should this be "logged in"?

**Comment by pmuir** [comment]
Otherwise I think this looks good. Sande can you test?

**Comment by pmuir** [comment]
I've also made a series of fixes here and merged.

Lukas, you should take a look at the commits I made on top of your work. Ping me if you are unsure why I've made these changes.

**Pull request closed without merging**